### PR TITLE
Add questions from study group

### PR DESCRIPTION
**Pull request by Xekyo**
• Retitles the question rubrics in each week per the titles of the materials
• Adds new questions to each week associated with materials by best guess

### CODE REVIEW

File: lightning-how-the-pieces-fit-together/README.md [hunk 1]
```diff
@@ -14,9 +14,43 @@
...
+### The Multihop Layer
+
+1. How can nodes that are not always online prevent loss of funds?
+1. What are the risks of forwarding below-dust payments?
+1. What is the relationship of the revocation keys for multiple subsequent commitment transactions? What would change if revocation keys were independent?
+1. How can you send a payment to a node without getting an invoice? What are the tradeoffs of this payment method?
+1. Why is a grace period added to the HTLC timeout?
+1. What are routing hints and why are they important for private channels? Do they leak privacy?
+
+### The other layers (base and transport)
```
Comment (adamjonas) [inline lightning-how-the-pieces-fit-together/README.md:46]: Probably remove if you don't have any questions to include

File: lightning-limitations/README.md [hunk 2]
```diff
@@ -17,7 +17,18 @@
...
+### Lightning Network 2.0
+
+### Major Limitations of Lightning
+
 1. What is the danger of being able to associate a base layer bitcoin node with the owner’s lightning node?
+1. What problem do anchor outputs resolve? Why don't they solve the pinning problem?
+
+### Spamming the Lightning Network
+
+1. What variables other than htlc_min_val and max_num_htlcs could be tweaked to make it harder or less attractive for an attacker try and lock up a channel?
```
Comment (adamjonas) [inline lightning-limitations/README.md:31]:
```suggestion
1. What variables other than `htlc_min_val` and `max_num_htlcs` could be tweaked to make it harder or less attractive for an attacker to try and lock up a channel?
```

File: payment-channels-htlcs/README.md [hunk 3]
```diff
@@ -15,30 +15,58 @@
...
+1. Why is revoking a transaction in bitcoin tricky? Can a user just take back a transaction?
+1. How is the commitment secret used to create the breach remedy transaction?
 
-### Timelocks
+### Bitcoin Timelocks Explained
 
 1. Why do CSV and CLTV rely on nLocktime or nSequence being set in the transaction spending the outputs when the script could check if the requirements are being met by itself?
 1. Can both CSV and CLTV be used in the same output and are there any known use cases for it?
 1. Is it fair to say UTXO-level timelocks are superior to transaction-level timelocks since they are baked directly into the protocol and allow more flexiblity and control? Are transaction-level timelocks necessary (Assuming there's a way to implement UTXO-level timelocks without relying on nLockTime/nSequence)?
+1. In what scenarios is OP_CLTV used in Lightning and in what scenarios is OP_CSV used?
```
Comment (adamjonas) [inline payment-channels-htlcs/README.md:49]: Not a strong opinion, but maybe put the OPs in code fences like `OP_CLTV`/`OP_CSV`

File: payment-channels-htlcs/README.md [hunk 4]
```diff
@@ -15,30 +15,58 @@
...
+### Visualizing HTLCs and the Lightning Network's Dirty Little Secret
 
 1. What do you think of the criticism that HTLCs don't work for micropayments?
+1. How do fees impact our ability to perform breach remedy?
+1. What factors may affect the minimum payment size for a multi-hop payment?
+
+### Lightning Network BOLT by BOLT
+
+1. What problem do commitment transactions solve? When a revoked commitment transaction is found, what outputs need to be resolved by the penalty transaction? What happens to the HTLCs?
+1. A node must be able to generate the <revocation_sig> for all prior revoked commitments. How can they be stored efficiently, what are the challenges?
```
Comment (adamjonas) [inline payment-channels-htlcs/README.md:69]:
```suggestion
1. A node must be able to generate the <revocation_sig> for all prior revoked commitments. How can they be stored efficiently? What are the challenges?
```

File: payment-channels-htlcs/README.md [hunk 5]
```diff
@@ -15,30 +15,58 @@
...
+1. What is a good value for to-self-delay for breach remedy and why?
```
Comment (adamjonas) [inline payment-channels-htlcs/README.md:70]:
```suggestion
1. What is a good value for to-self-delay for the breach remedy and why?
```

**Review by adamjonas — COMMENTED** [review]
These are good questions. Good job!

**Comment by adamjonas** [comment]
What about the requested changes?

File: lightning-how-the-pieces-fit-together/README.md (hunk 1 above)
Comment (Xekyo) [reply lightning-how-the-pieces-fit-together/README.md:46]: I'll comment out the unused sections.

File: lightning-limitations/README.md (hunk 2 above)
Comment (Xekyo) [reply lightning-limitations/README.md:31]: Done

**Pull request merged**